packages/fuel-indexer-tests/src/fixtures.rs

```diff
@@ -16,29 +17,174 @@ use fuels::{
     },
     signers::Signer,
 };
-use sqlx::{pool::Pool, Postgres};
+use rand::distributions::Alphanumeric;
+use rand::{thread_rng, Rng};
+use sqlx::{pool::Pool, PgConnection, Postgres};
+use sqlx::{Connection, Executor};
 use std::net::SocketAddr;
 use std::path::{Path, PathBuf};
+use std::str::FromStr;
 use tracing_subscriber::filter::EnvFilter;
 
 abigen!(Contract(
     name = "FuelIndexerTest",
     abi = "packages/fuel-indexer-tests/contracts/fuel-indexer-test/out/debug/fuel-indexer-test-abi.json"
 ));
 
-pub async fn postgres_connection_pool() -> Pool<Postgres> {
-    let config = DatabaseConfig::Postgres {
-        user: "postgres".into(),
-        password: "my-secret".into(),
-        host: "127.0.0.1".into(),
-        port: "5432".into(),
-        database: "postgres".to_string(),
-    };
-    match IndexerConnectionPool::connect(&config.to_string())
-        .await
-        .unwrap()
-    {
-        IndexerConnectionPool::Postgres(p) => p,
+pub struct TestPostgresDb {
+    pub db_name: String,
+    pub url: String,
+    pub pool: Pool<Postgres>,
+    server_connection_str: String,
+}
+
+impl TestPostgresDb {
+    pub async fn new() -> Self {
+        // Generate a random string to serve as a unique name for a temporary database
+        let rng = thread_rng();
+        let db_name: String = rng
+            .sample_iter(&Alphanumeric)
+            .take(10)
+            .map(char::from)
+            .collect();
+
+        // The server connection string serves as a way to connect directly to the Postgres server.
+        // Example database URL: postgres://postgres:my-secret@localhost:5432
+        let connection_config: DatabaseConfig = match std::env::var("DATABASE_URL")
+            .expect("DATABASE_URL environment variable is not set")
+            .parse()
+        {
+            Ok(cc) => cc,
+            Err(e) => panic!("Could not parse DATABASE_URL: {e}"),
+        };
+        let server_connection_str = connection_config.to_string();
+
+        let DatabaseConfig::Postgres {
+            user,
+            password,
+            host,
+            port,
+            ..
+        } = connection_config;
+        let test_db_config = DatabaseConfig::Postgres {
+            user,
+            password,
+            host,
+            port,
+            database: db_name.clone(),
+        };
+
+        // Connect directly to the Postgres server and create a database with the unique string
+        let mut conn = match PgConnection::connect(server_connection_str.as_str()).await {
```

Comment:
@deekerno 
- For this entire `TestPostgresDb::new` function, I realize this is testing code (thus not as picky with it). 
- But as I'm looking over it, maybe we should use the same error pattern as we use in the service code? 
- Meaning, in this test code we're using a lot of `expect` and `panic`. I'm thinking maybe we can just `?` a lot of this stuff?
  - Maybe with some `TestError` enum that uses `thiserror` similar to how `IndexerError` does? Thoughts?
    - Thinking is, since this PR doesn't really change _a ton_ about how the testing works (we still have to use a single-threaded setup), a real benefit/side-effect of this PR can be cleaning up our testing setup

Reply:
I'm down for that. I was doing some work to get multi-threading to work and also came to the conclusion that we could do some cleanup in places. I'll work on this with the other suggestions, and I'll ping for re-review.